<launch>
    <!-- USB config-->
    <arg default="/dev/ttyUSB0" name="uart_port"></arg>
    <arg default="/dev/ttyUSB1" name="data_port"></arg>

    <!-- Topic names-->

    <arg default="/gtec/mmwave/radar" name="publish_radar_topic"></arg>
    <arg default="/gtec/mmwave/cloud" name="publish_cloud_topic"></arg>
    <arg default="/gtec/mmwave/target" name="publish_target_topic"></arg>
    <arg default="/gtec/mmwave/cartesian" name="publish_cartesian_cloud_topic"></arg>

    <!-- Radar device config-->

    <arg default="$(find gtec_mmwave_reader)/example_radar_configs/tracking_people_6m_6843ISK.cfg" name="config_file_path"></arg>
    
    <arg default="0" name="radar_pos_x"></arg>
    <arg default="0" name="radar_pos_y"></arg>
    <arg default="1.741" name="radar_pos_z"></arg>
    <arg default="0" name="radar_yaw"></arg>
    <arg default="0" name="radar_pitch"></arg>
    <arg default="-0.22" name="radar_roll"></arg>

    <!-- Nodes -->

    <node name="IWR6843ISKPeopleCountingReader" output="screen" pkg="gtec_mmwave_reader" type="IWR6843ISK_people_counting_reader.py">
        <param name="uart_port" value="$(arg uart_port)"></param>
        <param name="data_port" value="$(arg data_port)"></param>
        <param name="config_file_path" value="$(arg config_file_path)"></param>
        <param name="publish_radar_topic" value="$(arg publish_radar_topic)"></param>
        <param name="publish_cloud_topic" value="$(arg publish_cloud_topic)"></param>
        <param name="publish_target_topic" value="$(arg publish_target_topic)"></param>
        <param name="sensor_height" value="$(arg radar_pos_z)"></param>
        <param name="elev_tilt" value="$(eval -1. * arg('radar_roll'))"></param>
    </node>

    <node pkg="tf" 
        type="static_transform_publisher" 
        name="odom_to_radar" 
        args="$(arg radar_pos_x) $(arg radar_pos_y) $(arg radar_pos_z) $(arg radar_yaw) $(arg radar_pitch) $(arg radar_roll) odom radar 30" />

    <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0 0 0 0 0 0 map odom 30" />

    <node pkg="pose_publisher" type="pose_publisher" name="radar_pose">
            <param name="map_frame" value="odom"></param>
            <param name="base_frame" value="radar"></param>
    </node>

    <node name="IWR6843ISKPolarToCartesian" output="screen" pkg="gtec_mmwave_reader" type="IWR6843ISK_polar_to_cartesian.py">
        <param name="radar_topic" value="$(arg publish_radar_topic)"></param>
        <param name="publish_cloud_topic" value="$(arg publish_cartesian_cloud_topic)"></param>>
    </node>

</launch>